hooks/basic/collector.py

```diff
@@ -214,12 +216,84 @@ def on_context_changed(self, settings, item):
         work_path_template = item_info["work_path_template"]
 
         # If defined, add the work_path_template to the item's properties
+        # set/evaluate the correct work_path_template
+        # First check if a work_path_template has been defined
+        if work_path_template:
+            work_tmpl = self.parent.get_template_by_name(work_path_template)
+
+            # template is invalid.
+            if not work_tmpl:
```

Comment:
Isn't that what we are trying to fix here? That if a template is invalid in the config (eg. tk-shell_project_work_file) would be resolved with a correct value using template_from_path? If we are trying to ingest/publish a file from home dir it would resolve to a context corresponding to where the publisher was launched from in the show.

Either I can create a new set of settings for standalone publisher in config which doesn't have {engine_name} in the templates. Or I can completely by-pass it with a try-except on hitting such error and try to fit the path to a known template?

Reply:
If the plugin is misconfigured, it should error. The solution here is to remove those template paths in the configuration and replace them with ones that work, or solely rely on template_from_path. Keep in mind that since most of our work path templates point at DCC-specific subdirs, i.e. nuke, maya, etc, you're going to have a hard time defining generic work path templates for the standalone publisher that work across all workflows. This is why I was suggesting to create a standalone publisher configuration per department instead, where you will always know where those work files are being stored.

Reply:
So, I will make this change in standalone publisher config and remove work_path_template from the settings, so that it will solely rely on template_from_path.